src/natcap/invest/spec.py

```diff
@@ -2141,8 +2147,8 @@ def setup(self, args, taskgraph_key='taskgraph_cache'):
         return args, file_registry, graph
 
     def execute(self, args, create_logfile=False, log_level=logging.NOTSET,
-            generate_metadata=False, save_file_registry=False,
-            check_outputs=False):
+                generate_metadata=False, save_file_registry=False,
+                check_outputs=False, generate_report=False):
```

Comment:
Is there ever a situation where we don't want to generate a report for a model that does have a reporting module?

Reply:
Probably not? I would be in favor of changing all of these to `True` by default. Except for `check_outputs`.

Reply:
In that case, would it make sense to determine whether to generate the report within this function, ie. `if self.reporter`, rather than passing it in as a flag?

Reply:
Well, I think we must have decided at one point that it made sense to have all of these as user-facing options. I don't see the report as different from the metadata or file registry in that respect, so I was just following the precedent we already set.